packages/core/src/App/Components/Routes/route-with-sub-routes.jsx

```diff
@@ -30,6 +31,7 @@ const RouteWithSubRoutes = route => {
             result = (
                 <React.Fragment>
                     {has_default_subroute && pathname === route.path && <Redirect to={default_subroute.path} />}
+                    {route.prompt_before_leave && <RoutePromptDialog condition={route.prompt_before_leave} />}
```

Comment:
Do you still need to check `route.prompt_before_leave` here even though `<Prompt />`'s `when` prop is already checking the condition?